### PR TITLE
Update README to aligh with change #450 and update ChangeLog for housekeeping pr #442

### PR DESCRIPTION
**Pull request by ldai1**
**Issue #:** 

**Description of changes:**
* Modify README to align with the latest code change #450 
* Add ChangeLog for housekeeping pr #442 (include changelog for bumping the react version)
* Add missing space and backticks for ChangeLog.

**Testing**
1. Have you successfully run `npm run build:release` locally? Yes

2. How did you test these changes? Manually test

3. If you made changes to the component library, have you provided corresponding documentation changes? Yes

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -11,15 +11,22 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
 ### Fixed
 
+- Fix incorrect type definitions in date formatting.
+- Fix the Jest configuration to not use the deprecated 'tsConfig' spelling of 'tsconfig'.
 - Fix removeEventListener bug for WithTooltip.
 - Reset `isAudioOn` state to `true` when `LocalAudioOutputProvider` unmounts.
 
 ### Added
 
+- - Added some key properties to 'ModalButtonGroup', which quiets some React warnings.
```
Comment (devalevenkatesh) [inline CHANGELOG.md:21]: Nit: Could you please remove extra "-" in front of the changlog line and backtick ModalButtonGroup, 'ModalButtonGroup' to `ModalButtonGroup` ?

File: CHANGELOG.md [hunk 2]
```diff
@@ -11,15 +11,22 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
 - Added an optional prop `onPopOverClick` to pass a callback function to the `PopOver` UI component. This callback will be called when the `PopOver` UI component is clicked.
 - Add `--no-fail-on-empty-changeset` flag in deploy script to not fail for empty changeset
 
 ### Changed
+- Bumped react and react-dom version to 17.
+- Updated versions of testing-library family of packages
```
Comment (devalevenkatesh) [inline CHANGELOG.md:27]: Nit: Could you please add a "." at the end of sentence here?

**Review by devalevenkatesh — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (ldai1) [reply CHANGELOG.md:21]: Good catch. Will remove.

File: CHANGELOG.md (hunk 2 above)
Comment (ldai1) [reply CHANGELOG.md:27]: Yes, will do.

File: scripts/release.js [hunk 3]
```diff
@@ -156,6 +156,15 @@ logger.warn(`
 
 shouldContinuePrompt();
 
+// Creates a new React test app, install the component app as dependency and build the test app to ensure no errors or build issues
```
Comment (devalevenkatesh) [inline scripts/release.js:159]: Would be helpful to have this as a separate PR since it is different from housekeeping one.

File: CHANGELOG.md [hunk 4]
```diff
@@ -11,15 +11,23 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ### Fixed
 
+- Fix incorrect type definitions in date formatting.
+- Fix the Jest configuration to not use the deprecated 'tsConfig' spelling of 'tsconfig'.
```
Comment (devalevenkatesh) [inline CHANGELOG.md:15]: Could you please backtick 'tsConfig' here?

File: CHANGELOG.md [hunk 5]
```diff
@@ -11,15 +11,23 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
 - Fix removeEventListener bug for WithTooltip.
 - Reset `isAudioOn` state to `true` when `LocalAudioOutputProvider` unmounts.
 
 ### Added
 
+- Added a step in release script to create a test react app, install latest version of component app as dependency and build the test app.
+- Added some key properties to 'ModalButtonGroup', which quiets some React warnings.
```
Comment (devalevenkatesh) [inline CHANGELOG.md:22]: Could we backtick 'ModalButtonGroup'?

File: CHANGELOG.md [hunk 6]
```diff
@@ -11,15 +11,23 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
 - Added an optional prop `onPopOverClick` to pass a callback function to the `PopOver` UI component. This callback will be called when the `PopOver` UI component is clicked.
-- Add `--no-fail-on-empty-changeset` flag in deploy script to not fail for empty changeset
+- Added `--no-fail-on-empty-changeset` flag in deploy script to not fail for empty changeset.
 
 ### Changed
+- Bumped react and react-dom version to 17.
+- Updated versions of testing-library family of packages.
+- Corrected the detection of 'yesterday' in said code.
+- Wraped actions in tests with 'act', as React requests.
```
Comment (devalevenkatesh) [inline CHANGELOG.md:30]: Could we backtick 'act'?

File: scripts/release.js [hunk 7]
```diff
@@ -156,6 +156,15 @@ logger.warn(`
...
+process.chdir(path.join(__dirname, '../..'));
+spawnOrFail('npx', ['create-react-app test-app']);
+process.chdir(path.join(__dirname, '../../test-app'));
+spawnOrFail('npm', [`install amazon-chime-sdk-js@${updatedSdkVersion}`]);
+spawnOrFail('npm', ['run build']);
```
Comment (devalevenkatesh) [inline scripts/release.js:164]: I think we also have to catch any dependency warnings if reported here, right?

**Review by devalevenkatesh — CHANGES_REQUESTED** [review]

**Review by devalevenkatesh — APPROVED** [review]
